### PR TITLE
Refactor WorkerRegisterContext

### PR DESCRIPTION
**Pull request by beinan**
### What changes are proposed in this pull request?

Refactor WorkerRegisterContext

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

**Review by rongrong — APPROVED** [review]

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 1]
```diff
@@ -442,7 +442,7 @@ public void heartbeat() {
         String msg = String.format(
             "ClockTime: %d, LastActivityTime: %d. Worker %d register stream hanging for %sms!"
                 + " Tune up %s if this is undesired.",
-            clockTime, lastActivityTime, context.mWorker.getId(), staleTime,
+            clockTime, lastActivityTime, context.getWorkerInfo().getId(), staleTime,
```
Comment (rongrong) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java:445]: There is a `context.getWorkerId()` btw. It's less verbose to directly use that. Or maybe remove that function and change all call sites to context.getWorkerInfo().getId()`.

Comment (beinan) [reply core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java:445]: Sounds good, I will post another pr to remove the getWorkerId, thx!

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**